bats/tests/snapshots/create-use-snapshot.bats

```diff
@@ -0,0 +1,91 @@
+load '../helpers/load'
+
+local_setup() {
+    if is_windows; then
+        skip "snapshots test not applicable on Windows"
+    fi
+    SNAPSHOT=moby-nginx-snapshot01
+}
+
+@test 'factory reset and delete the snapshot if it exists' {
+    factory_reset
+    run get_snapshot_id_from_name "$SNAPSHOT"
+    assert_success
+    if [[ -n $output ]]; then
+        rdctl snapshot delete "$output"
+    fi
+}
+
+@test 'start up in moby' {
+    RD_CONTAINER_ENGINE=moby
+    start_kubernetes
+    wait_for_container_engine
+    wait_for_apiserver
+}
+
+@test 'push an nginx pod and verify' {
+    kubectl run nginx --image=nginx:latest --port=8080
+    try --max 48 --delay 5 running_nginx
+}
+
+@test 'shutdown, make a snapshot, and clear everything' {
+    rdctl shutdown
+    rdctl snapshot create "$SNAPSHOT"
+    run rdctl snapshot list
+    assert_success
+    assert_output --partial "$SNAPSHOT"
+    rdctl factory-reset
+}
+
+@test 'startup, verify using new defaults' {
+    RD_CONTAINER_ENGINE=containerd
+    start_kubernetes
+    wait_for_container_engine
+    wait_for_apiserver
+    run rdctl api /settings
+    assert_success
+    run jq_output .containerEngine.name
+    assert_success
+    assert_output --partial containerd
+    run kubectl get pods -A
+    assert_success
+    refute_output --regexp 'default.*nginx.*Running'
+}
+
+# This should be one long test because if `snapshot restore` fails there's no point starting up
+@test 'shutdown, restore, restart and verify snapshot state' {
+    local snapshotID
+    rdctl shutdown
+    run get_snapshot_id_from_name "$SNAPSHOT"
+    assert_success
+    refute_output ""
+    snapshotID="$output"
+    run rdctl snapshot restore "$snapshotID"
+    assert_success
+    refute_output --partial $"failed to restore snapshot \"$snapshotID\""
+
+    # Circumvent having start_kubernetes => start_container_engine set all the defaults
```

Comment:
Rather than introducing code like this for every case that needs it, wouldn't it be better to add a helper function that simply starts RD without touching settings at all? Though we may want to set some things by default. Either way, IMO we should have a helper function for this.

Reply:
It took me a while to write `launch_application`, but I got it. Yes, better to have the differences between dev and release-launching contained to one place.